megamek/src/megamek/client/bot/CEntity.java

```diff
@@ -668,7 +668,7 @@ public void characterize() {
         }
 
         if (entity instanceof Infantry) {
-            if (!(entity instanceof BattleArmor)) {
+            if (entity.isConventionalInfantry()) {
```

Comment:
Under the hood, this introduces a redundant comparison.

Reply:
You'd prefer
```
if (entity.isConventionalInfantry()) {
} else if (entity instanceof BattleArmor) {
```
?

Because... performance-wise this is an improvement from the current check.

Reply:
Looked under the hood at isConventionalInfantry(), and turns out it doesn't do what I thought it did. So never mind.